lib/wasi/src/syscalls/wasix32.rs

```diff
@@ -2,6 +2,12 @@
 use crate::{WasiEnv, WasiError, WasiState, WasiThread};
 use wasmer::{FunctionEnvMut, Memory, Memory32, MemorySize, StoreMut, WasmPtr, WasmSlice};
 use wasmer_wasi_types::*;
```

Comment:
Do we still need `wasmer_wasi_types` ? Why?

Reply:
Not all types can be ported to WIT, especially not types that use generics.

Reply:
Kind of make sense... 
I see the file: https://github.com/wasmerio/wasmer/blob/wasi-types-generation-2/lib/wasi-types/src/lib.rs

I think it might make sense to move that into the `wasmer-wasi` repo.
Although I see some types that might be feasible to move to wit?
* `subscription::EventType`
* `net::*` (most of the net types?)


I also think the bus type names would make sense to have different types if it's 32 or 64 bits (so it doesn't need to be generic).
cc @john-sharratt

Reply:
@syrusakbary The `subscription` module is portable (I missed that somehow), but net uses arrays, which WIT does not support (that's why I couldn't port it).

Reply:
Ok. Let's port the subscription module. And for the rest of things that can't be ported let's just have a `types.rs` files inside `wasmer-wasi` instead of a new crate